shell/run_types.ts

```diff
@@ -0,0 +1,36 @@
+export type RunType = 'piped' | 'streamed';
+
+interface ExtensionRunOptionsExtension {
```

Comment:
Small nit: The name `ExtensionRunOptionsExtension` feels like maybe it has one too many terms? Would `RunOptionsExtension` be consistent with what you are intending here?

Reply:
Not sure how I ended up with that. 🙃